### PR TITLE
Suppress errors when a request contains no job id

### PR DESCRIPTION
**Pull request by jmuharsky**
This has been an issue for error conditions, as well as certain cases with text/cloudsql widgets.

Demo is at https://27-1-dot-perfkit-explorer.googleplex.com

### CODE REVIEW

File: client/components/widget/query/query-result-data-service.js [hunk 1]
```diff
@@ -233,19 +233,29 @@ QueryResultDataService.prototype.fetchResults = function(widget) {
         let rows = response.data.totalRows;
         let size = response.data.totalBytesProcessed;
         let time = response.data.elapsedTime;
-        let job = response.data.jobReference.jobId;
+        let jobReference = response.data.jobReference;
 
-        widget.state().datasource.job_id = job;
+        if (goog.isDefAndNotNull(jobReference)) {
```
Comment (ivansmf) [inline client/components/widget/query/query-result-data-service.js]: You need to know whether this is a BigQuery request or a CloduSQL request to eat this error. What happens if this is indeed a BigQuery request without a job id?

Comment (jmuharsky) [inline client/components/widget/query/query-result-data-service.js]: Then the jobid doesn't get recorded.  And there are a few edge error cases we've found where this occurs -- BigQuery returns, but there isn't a jobid.

At a larger scale, this class needs to be factored out for proper extensibility, but that hasn't been designed yet, and it will be in a separate PR.

Comment (ivansmf) [inline client/components/widget/query/query-result-data-service.js]: Please add a TODO for the refactoring, otherwise this is fine - I will pull it in.
